app/src/lib/stores/app-store.ts

```diff
@@ -5271,18 +5272,47 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
   public async _checkoutPullRequest(
     repository: Repository,
-    pullRequest: PullRequest
+    pullRequest: IAPIPullRequest | PullRequest
   ): Promise<void> {
+    let prNumber: number | null = null
+    let ownerLogin: string | null = null
+    let head: {
+      cloneURL: string
+      ref: string
+    } | null = null
+
+    if (pullRequest instanceof PullRequest) {
+      if (pullRequest.head.gitHubRepository.cloneURL === null) {
+        return
+      }
+
+      prNumber = pullRequest.pullRequestNumber
+      ownerLogin = pullRequest.author
+      head = {
+        cloneURL: pullRequest.head.gitHubRepository.cloneURL,
+        ref: pullRequest.head.ref,
+      }
+    } else {
+      if (pullRequest.head.repo === null) {
+        return
+      }
+
+      prNumber = pullRequest.number
+      ownerLogin = pullRequest.user.login
+      head = {
+        cloneURL: pullRequest.head.repo.clone_url,
+        ref: pullRequest.head.ref,
+      }
+    }
```

Comment:
Not particularly proud of this piece of logic, but it allows the `_checkoutPullRequest()` method to accept both a `PullRequest` object and an `IAPIPullRequest` one.

I can also move this logic into a separate method to try to keep this method cleaner.